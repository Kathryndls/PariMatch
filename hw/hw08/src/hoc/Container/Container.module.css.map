{
    "version": 3,
    "mappings": "AAAA,AAAA,UAAU,CAAC;EACR,MAAM,EAAE,MAAM;EACd,OAAO,EAAE,MAAM;EACf,SAAS,EAAE,MAAM;EACjB,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,KAAK;CACf;;AAED,MAAM,EAAE,SAAS,EAAE,MAAM;EACtB,AAAA,UAAU,CAAC;IACR,KAAK,EAAE,MAAM;GACf;;;AAGJ,MAAM,EAAE,SAAS,EAAE,MAAM;EACtB,AAAA,UAAU,CAAC;IACR,KAAK,EAAE,KAAK;GACd;;;AAGJ,MAAM,EAAE,SAAS,EAAE,KAAK;EACrB,AAAA,UAAU,CAAC;IACR,KAAK,EAAE,KAAK;GACd;;;AAGJ,MAAM,EAAE,SAAS,EAAE,KAAK;EACrB,AAAA,UAAU,CAAC;IACR,KAAK,EAAE,KAAK;GACd;;;AAGJ,MAAM,EAAE,SAAS,EAAE,KAAK;EACrB,AAAA,UAAU,CAAC;IACR,KAAK,EAAE,IAAI;IACX,OAAO,EAAE,GAAG;GACd",
    "sources": [
        "Container.module.scss"
    ],
    "names": [],
    "file": "Container.module.css"
}