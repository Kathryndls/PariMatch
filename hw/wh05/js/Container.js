export class Container {
  constructor(item) {
    this.item = item;
  }

  render() {}

  renderWrapper() {
    const wrapper = document.createElement("div");
    wrapper.classList.add("wrapper");

    const header = document.createElement("div");
    header.classList.add("item-header");

    wrapper.append(header);

    return {
      wrapper,
      header
    };
  }

  renderDeleteButton() {
    const deleteBtn = document.createElement("button");
    deleteBtn.classList.add("delete-btn");
    deleteBtn.textContent = "DELETE";

    return deleteBtn;
  }

  renderSortButton() {
    const sortBtn = document.createElement("button");
    sortBtn.classList.add("sort-btn");
    sortBtn.textContent = "SORT";

    return sortBtn;
  }

  renderTitle() {
    const title = document.createElement("h2");
    title.classList.add("item-title");
    title.textContent = this.item.title;
    return title;
  }
}

// TITLE
export class Title extends Container {
  constructor(item) {
    super(item);
    this.item = item;
    this.title = item.title;
  }

  render() {
    const root = document.querySelector("#root");
    const {wrapper, header } = this.renderWrapper();
    const title = this.renderTitle();
    const deleteBtn = this.renderDeleteButton();
    const sortBtn = this.renderSortButton();

    header.append(title, deleteBtn, sortBtn);

    this.item.fields.forEach((field) => {
      const input = document.createElement("input");

      input.classList.add("item-text-input", "item-input");
      input.setAttribute("type", field.type);
      input.setAttribute("placeholder", field.label);

      wrapper.append(input);
    });

    root.append(wrapper);
  }
}

// RICH TEXT
export class RichText extends Container {
  constructor(item) {
    super(item);
    this.item = item;
    this.title = item.title;
  }

  render() {
    const root = document.querySelector("#root");
    const { wrapper, header } = this.renderWrapper();
    const title = this.renderTitle();
    const deleteBtn = this.renderDeleteButton();
    const sortBtn = this.renderSortButton();

    header.append(title, deleteBtn, sortBtn);

    this.item.fields.forEach((field) => {
      const textArea = document.createElement("textarea");

      textArea.classList.add("item-rich-text-input", "item-input");
      textArea.setAttribute("type", field.type);
      textArea.setAttribute("placeholder", field.label);

      wrapper.append(textArea);
    });

    root.append(wrapper);
  }
}

// ACTION_BUTTON
export class ActionButton extends Container {
  constructor(item) {
    super(item);
    this.item = item;
    this.title = item.title;
  }

  render() {
    const root = document.querySelector("#root");
    const { wrapper, header } = this.renderWrapper();
    const title = this.renderTitle();
    const deleteBtn = this.renderDeleteButton();
    const sortBtn = this.renderSortButton();

    header.append(title, deleteBtn, sortBtn);

    this.item.fields.forEach((field) => {
      const input = document.createElement("input");

      input.classList.add("item-action-text-input", "item-input");
      input.setAttribute("type", field.type);
      input.setAttribute("placeholder", field.label);

      wrapper.append(input);
    });

    root.append(wrapper);
  }
}

// IMAGE
export class Image extends Container {
  constructor(item) {
    super(item);
    this.item = item;
    this.title = item.title;
  }

  render() {
    const root = document.querySelector("#root");
    const { wrapper, header } = this.renderWrapper();
    const title = this.renderTitle();
    const deleteBtn = this.renderDeleteButton();
    const sortBtn = this.renderSortButton();

    header.append(title, deleteBtn, sortBtn);

    this.item.fields.forEach((field) => {
      const inputWrapper = document.createElement("div");
      const input = document.createElement("input");
      const label = document.createElement("label");

      inputWrapper.classList.add("item-input-wrapper");

      input.classList.add("item-text-input", "item-input");
      input.setAttribute("type", "file");

      label.classList.add("item-image-label");
      label.textContent = field.label;

      inputWrapper.append(label, input);
      wrapper.append(inputWrapper);
    });

    root.append(wrapper);
  }
}

// GALLERY
export class Gallery extends Container {
  constructor(item) {
    super(item);
    this.item = item;
    this.title = item.title;
  }

  render() {
    const root = document.querySelector("#root");
    const { wrapper, header } = this.renderWrapper();
    const title = this.renderTitle();
    const deleteBtn = this.renderDeleteButton();
    const sortBtn = this.renderSortButton();

    header.append(title, deleteBtn, sortBtn);

    this.item.fields.forEach((field) => {
      const inputWrapper = document.createElement("div");
      const input = document.createElement("input");
      const label = document.createElement("label");

      inputWrapper.classList.add("item-input-wrapper");

      input.classList.add("item-text-input", "item-input");
      input.setAttribute("type", "file");

      label.classList.add("item-image-label");
      label.textContent = field.label;

      inputWrapper.append(label, input);
      wrapper.append(inputWrapper);
    });

    root.append(wrapper);
  }
}
