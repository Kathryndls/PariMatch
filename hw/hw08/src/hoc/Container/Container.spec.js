import React from "react";
import Container from "./Container";

const setUp = (props) => shallow(<Container {...props} />);
const setUpRender = (props) => render(<Container {...props} />);

describe('should render Container HOC', () => {
   let component;
   beforeEach(() => {
      component = setUp();
   })

   it ('should have a Container class', () => {
      const wrapper = component.find('.Container');
      expect(wrapper.length).toBe(1);
   });

   it ('should have a div tag', () => {
      const wrapper = component.find('div');
      expect(wrapper.length).toBe(1);
   })
});

describe('Container HOC', () => {
   it ('should render Container HOC', () => {
      const component = setUpRender();
      expect(component).toMatchSnapshot();
   })
});