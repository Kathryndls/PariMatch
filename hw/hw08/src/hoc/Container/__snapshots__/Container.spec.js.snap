// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container HOC should render Container HOC 1`] = `
<div
  class="Container"
/>
`;
